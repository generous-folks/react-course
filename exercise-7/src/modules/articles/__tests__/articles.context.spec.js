import React from 'react';

import { shallow } from 'enzyme';

import {
  useArticles,
  useArticlesState,
  useArticlesDispatch,
  ArticlesProvider,
} from '../articles.context';

describe('articles.context', () => {
  describe('<ArticlesProvider />', () => {
    it('should render nested providers', () => {
      expect(
        shallow(
          <ArticlesProvider>
            <div />
          </ArticlesProvider>,
        ),
      ).toMatchSnapshot();
    });

    it('should have property value set to the articles state articles', () => {
      expect(
        shallow(
          <ArticlesProvider>
            <div />
          </ArticlesProvider>,
        )
          .find('ContextProvider')
          .first()
          .prop('value'),
      ).toMatchObject({ articles: [] });
    });
  });

  describe('useArticlesDispatch', () => {
    it('should be defined', () => {
      expect(typeof useArticlesDispatch).toBe('function');
    });
  });

  describe('useArticlesState', () => {
    it('should be defined', () => {
      expect(typeof useArticlesState).toBe('function');
    });
  });

  describe('useArticles', () => {
    it('should be defined', () => {
      expect(typeof useArticles).toBe('function');
    });
  });
});
