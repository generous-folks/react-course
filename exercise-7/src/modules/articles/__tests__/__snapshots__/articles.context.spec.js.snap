// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articles.context <ArticlesProvider /> should render nested providers 1`] = `
<ContextProvider
  value={
    Object {
      "articles": Array [],
    }
  }
>
  <ContextProvider
    value={[Function]}
  >
    <div />
  </ContextProvider>
</ContextProvider>
`;
